api/setting.go

```diff
@@ -57,17 +56,11 @@ func (s *Server) SettingSet(w http.ResponseWriter, req *http.Request) error {
 	}
 
 	name := mux.Vars(req)["name"]
-	sName := types.SettingName(name)
-
-	if sName == types.SettingNameBackupTarget {
-		// additional check whether have $ or , have been set in BackupTarget
-		regStr := `[\$\,]`
-		reg := regexp.MustCompile(regStr)
-		findStr := reg.FindAllString(setting.Value, -1)
-		if len(findStr) != 0 {
-			return fmt.Errorf("fail to set settings with invalid BackupTarget %s, contains %v", setting.Value, strings.Join(findStr, " or "))
-		}
+	err := s.m.SettingValidation(name, setting.Value)
```

Comment:
We can follow the pattern on other API calls, that combined the validation with `Set`. So just combine this with `m.CreateOrUpdateSetting`